remme/tp/node_account.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018 REMME
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+
+import logging
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+
+from remme.protos.node_account_pb2 import (
+    NodeAccount, NodeAccountMethod, NodeAccountInternalTransferPayload,
+)
+
+from remme.shared.forms import NodeAccountInternalTransferPayloadForm
+from remme.shared.constants import Events, EMIT_EVENT
+
+from .basic import (
+    PB_CLASS, PROCESSOR, VALIDATOR, BasicHandler, get_data, get_multiple_data
+)
+from remme.settings.helper import _get_setting_value
+LOGGER = logging.getLogger(__name__)
+
+FAMILY_NAME = 'node_account'
+FAMILY_VERSIONS = ['0.1']
+
+
+def get_node_account_by_address(context, address):
+    account = get_data(context, NodeAccount, address)
+    if account is None:
+        return NodeAccount()
+    return account
+
+
+class NodeAccountHandler(BasicHandler):
+    def __init__(self):
+        super().__init__(FAMILY_NAME, FAMILY_VERSIONS)
+
+    def get_state_processor(self):
+        return {
+            NodeAccountMethod.TRANSFER_FROM_FROZEN_TO_UNFROZEN: {
+                PB_CLASS: NodeAccountInternalTransferPayload,
+                PROCESSOR: self._transfer_from_frozen_to_unfrozen,
+                VALIDATOR: NodeAccountInternalTransferPayloadForm,
+            }
+        }
+
+    def _transfer_from_frozen_to_unfrozen(self, context, node_account_public_key, internal_transfer_payload):
+        node_account_address = self.make_address_from_data(node_account_public_key)
+
+        node_account = get_node_account_by_address(context, node_account_address)
+
+        if node_account.reputation.frozen < 250_000:
```

Comment:
Move `250_000` to the constant that will describe business-logic.

Example:
- I have a broken car, I want to fix the car,
- the minimum amount of dollars I should pay to fix the car is `10.000$`,
- so in the code, I will create a constant like `MINIMUM_AMOUNT_OF_DOLLARS_TO_FIX_THE_CAR`.

Reply:
Done.